### PR TITLE
Remove outdated practices from documentation

### PR DESCRIPTION
**Pull request by jsma**
This pull request removes references to deprecated functionality (South, django-mptt), updates the CMSApp docs to use current naming conventions rather than the old deprecated names, along with a few other minor changes/fixes.

### CODE REVIEW

**Comment by FinalAngel** [comment]
amazing job @jsma can you review this @evildmp ?

**Comment by evildmp** [comment]
@jsma This is great, thanks.

There's going to be a patch https://github.com/divio/django-cms/pull/5399 landing shortly that will affect much of the above, so it will shortly need to be rebased.

Also, please squash this into a single commit when it's ready.

I'll leave some specific comments inline.

**Comment by jsma** [comment]
Thanks for the pointer to #5399! All of the work in this PR was the yak shaving I did on the way to making my own updates to the apphooks documentation ;) I'll take a look at your PR, it may address my concerns.

I'll squash this if there are no objections to the changes I've made. I left it as several micro commits in case any particular type of change would be vetoed.

**Comment by jsma** [comment]
I've rebased and squashed. I removed my commit related to CMSApp._urls/CMSApp._menus, as there appears to be a lot of inconsistency in the docs and code related to this, so I'll open a separate PR so this PR can focus on minor fixes/updates to documentation only.

**Comment by jsma** [comment]
Anything else I need to do to get this ready for merging? Are there related changes that need to be made simultaneously with the documentation updates using `__str__()`?

**Comment by FinalAngel** [comment]
@evildmp is this ready to go?

File: docs/how_to/namespaced_apphooks.rst [hunk 1]
```diff
@@ -153,7 +153,7 @@ Create the FAQ ``Entry`` model
 
         objects = AppHookConfigManager()
 
-        def __unicode__(self):
+        def __str__(self):
```
Comment (evildmp) [inline docs/how_to/namespaced_apphooks.rst]: Can you update this to:  ``def __str__(self):  # use __unicode__() if you are using Python 2``? - applies to other instances too, of course.

**Comment by jsma** [comment]
Done! I also rebased again so everything is up to date.

**Comment by FinalAngel** [comment]
@evildmp please merge :)

**Comment by evildmp** [comment]
Rebased and committed in https://github.com/divio/django-cms/commit/600f50044da5a36f889acb719ab1cfb9917415db.

Thanks @jsma.

**Pull request closed without merging**